decentraland/components/presentational/BuyLand.js

```diff
@@ -14,31 +14,28 @@ export default function BuyLand(props) {
     landForSale,
     onBuy,
     accountCreationStatus,
-    accountCreationAction,
+    accountCreationActions,
   } = props;
 
+  const accountCreationAction = accountCreationActions[accountCreationStatus];
+
   const waitingForAccountSetup =
     accountCreationStatus !== NOT_STARTED &&
     accountCreationStatus !== READY_TO_USE;
 
+  const onPress = waitingForAccountSetup ? () => {} : onBuy;
+  const buttonDisabled = waitingForAccountSetup ? true : false;
+
   return (
     <View style={styles.container}>
       <LandForSale landForSale={landForSale} />
-      {waitingForAccountSetup ? (
-        <React.Fragment>
-          <View style={styles.buttonView}>
-            <Button title="Buy" disabled={true} onPress={() => {}} />
-          </View>
-          <AccountCreationProgress
-            status={accountCreationStatus}
-            action={accountCreationAction}
-          />
-        </React.Fragment>
-      ) : (
-        <View style={styles.buttonView}>
-          <Button title="Buy" onPress={onBuy} />
-        </View>
-      )}
+      <View style={styles.buttonView}>
+        <Button title="Buy" disabled={buttonDisabled} onPress={onPress} />
```

Comment:
> I've just got a BUG on this PR:
> Since we are getting the list of users' land from blockchain during `MyAssetsScreen.componentDidMount` what is happening is:
> On the first time of the app: User signs up, buy land and before the purchase process finished on the blockchain, he is moved to the `MyAssetsScreen` that will get no assets from blockchain and the list will be empty.
> On the second use, the list will be loaded with the land bought on first use and so on...
> The expected behavior only will happen if the user access the `MyAssetsScreen` before purcharse any asset.
> I think will could move the user the the `LandForSaleListScreen` as before, and extract this change (move to MyLand) to already created issue #267

Assuming that you agree since you gave thumbs up to it. I've just undo the navigation change. I'll add more info to the #267 right now.

Reply:
Ops, it isn't the best place to do that comment.

Reply:
Ah ok yeah. We can close this thread and discuss back in the original comment thread / main PR thread.

Reply:
Done. Resolving this.